src/app/features/courses/courseCard.component.ts

```diff
@@ -0,0 +1,35 @@
+import { Component, Input } from "@angular/core";
+
+
+@Component({
+    selector: 'card',
+    templateUrl: './courseCard.component.html',
+    styleUrls: ['./courseCard.component.css']
+
+})
+
+export class CourseCardComponent {
+
+    _duration: string = '';
+    _editable: string = '';
+    isEditable: Boolean = false
+    @Input() title: string = '';
+    @Input() text?: string;
+    @Input() creationDate?: string;
+    @Input() authors?: string;
+    @Input()
+    set duration(duration: string) {
+        if (duration.length > 0)
+            this._duration = `${Math.trunc(Number(duration) / 60)}:${Number(duration) % 60} hours`;
```

Comment:
it is good approach to use get/set for Inputs but it would be better to use a pipe. In this case you would be able to reuse it

Reply:
and additianly it would perform better then a setter , please read about pure pipes and how it could be used for performance optimisations